File: elixir_gql/lib/elixir_gql/spotify/musica.ex
defmodule ElixirGql.Spotify.Musica do
  use Ecto.Schema
  import Ecto.Changeset

  schema "musicas" do
    field :artista, :string
    field :nome, :string
    many_to_many :playlists, ElixirGql.Spotify.Playlist, join_through: "playlists_musicas"

    timestamps()
  end

  @doc false
  def changeset(musica, attrs) do
    musica
    |> cast(attrs, [:nome, :artista])
    |> validate_required([:nome, :artista])
  end
end

File: ts_grpc/src/entity/Musica.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToMany } from 'typeorm';
import { Playlist } from './Playlist';

@Entity()
export class Musica {
  @PrimaryGeneratedColumn({ type: 'int' })
  ID!: number;

  @Column({ type: 'varchar', length: 100, nullable: true })
  Nome: string | undefined;

  @Column({ type: 'varchar', length: 100, nullable: true })
  Artista: string | undefined;

  @ManyToMany(() => Playlist, playlist => playlist.musicas)
  playlists: Playlist[] | undefined;
}
